game/scripts/vscripts/components/points/points.lua

```diff
@@ -17,5 +24,81 @@ function PointsManager:Init ()
   end
 
   -- set initial values for current scores
-  CustomNetTables:SetTableValue( "team_scores", "score", { radiant = 0, dire = 0 } )
+  CustomNetTables:SetTableValue( "team_scores", "score", { goodguys = 0,
+                                                           badguys = 0
+                                                         })
+
+  -- Start Thinking
+  Timers:CreateTimer(0, Dynamic_Wrap(PointsManager, "Think"))
+end
+
+function PointsManager:Think ()
+  DebugPrint("[points/PointsManager] Thinking..")
+
+  local interval = 5 -- when do we want to check for win conditions
+  local limit = CustomNetTables:GetTableValue("team_scores", "limit").value
+  local scores = CustomNetTables:GetTableValue("team_scores", "score")
+  local goodguys = scores.goodguys
+  local badguys = scores.badguys
+
+  DebugPrintTable(limit)
+  DebugPrintTable(scores)
+  DebugPrint("haveGoodguysWon: " .. tostring(PointsManager.haveGoodguysWon))
+  DebugPrint("haveBadguysWon: " .. tostring(PointsManager.haveBadguysWon))
+
+  PointsManager:SetPoints(PointsManager.goodguysName, 200)
+
+  if PointsManager.haveGoodguysWon or PointsManager.haveBadguysWon then
+    return
+  end
+
+  if goodguys >= limit then
+    PointsManager:onTeamWin(PointsManager.goodguysName)
+  elseif badguys >= limit then
+    PointsManager:onTeamWin(PointsManager.badguysName)
+  end
+
+  return interval
+end
+
+function PointsManager:onTeamWin (side)
```

Comment:
`onTeamWin` seems like an event listener for a team winning, such as
```lua
PointsManager:onTeamWin(function (team)
  -- do things on win because observer pattern is awesome
  HookersAndBlow:sendToWinner(team)
end)
```

perhaps name it `victoryHandler`?